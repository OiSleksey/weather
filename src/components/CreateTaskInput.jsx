import React, { Component } from 'react';
import PropTypes from 'prop-types';

class CreateTaskInput extends Component {
  state = {
    value: '',
  };

  handleChange = event => {
    this.setState({
      value: event.target.value,
    });
  };

  handleClick = () => {
    this.props.sendInputData(this.state);
    this.setState({
      value: '',
    });
  };

  render() {
    return (
      <div className="create-task">
        <input
          type="text"
          className="create-task__input"
          value={this.state.value}
          onChange={this.handleChange}
        />
        <button className="btn" onClick={this.handleClick}>
          Create
        </button>
      </div>
    );
  }
}

export default CreateTaskInput;

CreateTaskInput.propTypes = {
  sendInputData: PropTypes.func.isRequired,
};
