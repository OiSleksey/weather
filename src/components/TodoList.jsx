import React, { Component, useEffect } from 'react';
import ScreenExtension from './ScreenExtension';
import TaskList from './TaskList';
import CreateTaskInput from './CreateTaskInput';
import { connect } from 'react-redux';
import * as asyncAction from '../tasks/tasks.actions';
import {
  taskListSelector,
  sortingTasksSelector,
} from '../tasks/tasks.selector';
import { setOtherUrl } from '../scripts/dataBaseRequsts';
import PropTypes from 'prop-types';

const TodoList = props => {
  useEffect(() => {
    //Start assigning the first url in fetch Data Base
    setOtherUrl();
    //Start assigning the first date from Data Base
    props.getInfoTask();
  }, []);

  const taskList = sortingTasksSelector(props.tasks);
  return (
    <div className="todo-list">
      <ScreenExtension />
      <CreateTaskInput sendInputData={props.createNewTask} />
      <ul className="list">
        {taskList.map(task => (
          <TaskList
            key={task.id}
            {...task}
            setChecked={props.updateTask}
            deleteTask={props.deleteTask}
          />
        ))}
      </ul>
    </div>
  );
};

TodoList.propTypes = {
  getInfoTask: PropTypes.func.isRequired,
  createNewTask: PropTypes.func.isRequired,
  updateTask: PropTypes.func.isRequired,
  deleteTask: PropTypes.func.isRequired,
  tasks: PropTypes.array.isRequired,
};

const mapState = state => {
  return {
    tasks: taskListSelector(state),
  };
};

const mapDispatch = {
  getInfoTask: asyncAction.getTodoTasks,
  createNewTask: asyncAction.createTodoTask,
  updateTask: asyncAction.updateTodoTask,
  deleteTask: asyncAction.deleteTodoTask,
};
export default connect(mapState, mapDispatch)(TodoList);
