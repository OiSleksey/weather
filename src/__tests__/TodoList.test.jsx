import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import {
  getRequest,
  postRequest,
  putRequest,
  deleteRequest,
  setOtherUrl,
} from '../dataBaseRequsts';
import TodoList from '../TodoList';
import { createSerializer } from 'enzyme-to-json';
import CreateTaskInput from '../CreateTaskInput';
// import axios from 'axios';

//Configure snapshot (enzyme-to-json)
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

//Configure (@cfaester/enzyme-adapter-react-18)
configure({ adapter: new Adapter() });

// jest.mock('../CreateTaskInput', () => {
//   CreateTaskInput = jest.fn();
// });

// jest.mock('../TaskList', () => {
//   TaskList = jest.fn();
// });

// const testRequest = () => {
//   getRequest().then(res => console.log(res));
//   console.log('Jest TEST');
// };

// testRequest();

describe('<TodoList />', () => {
  jest.mock('../dataBaseRequsts', () => {
    const tasks = [
      {
        text: 'Task 1',
        done: true,
        id: '27 123vb',
      },
    ];
    return {
      getRequest: jest.fn(() => Promise.resolve()),
      postRequest: jest.fn(() => Promise.resolve()),
      putRequest: jest.fn(() => Promise.resolve()),
      deleteRequest: jest.fn(() => Promise.resolve()),
      setOtherUrl: jest.fn(() => Promise.resolve()),
    };
  });
  it('should test TaskList', () => {
    const wrapperComponent = shallow(<TodoList />);

    expect(wrapperComponent.find('.todo-list')).toMatchSnapshot();
  });

  //   it('should test TaskList 2 ', () => {
  //     const wrapperComponent = shallow(<TodoList />);

  //     expect(wrapperComponent.find('CreateTaskInput')).toMatchSnapshot();
  //   });
});

// // app.js
// export async function fetchData() {
//     const response = await fetch('https://example.com/data');
//   const data = await response.json();
//   return data;
//   }
//   // app.test.js
//   import { fetchData } from './app';

//   test('fetchData returns the expected data', async () => {
//   // Create a mock implementation of the fetch function that returns a resolved Promise
//   global.fetch = jest.fn(() => Promise.resolve({json: () => Promise.resolve({ name: 'John Doe', age: 30 })}))
//   // Call the fetchData function and verify the returned data
//   const data = await fetchData();

//   expect(data).toEqual({ name: 'John Doe', age: 30 });
//   // Verify that the fetch function was called with the correct URL
//   expect(global.fetch).toHaveBeenCalledWith('https://example.com/data');
// })
