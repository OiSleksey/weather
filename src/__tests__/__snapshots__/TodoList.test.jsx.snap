// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> should test TaskList 1`] = `
<div
  className="todo-list"
>
  <CreateTaskInput
    sendInputData={[Function]}
  />
  <ul
    className="list"
  />
</div>
`;
